src/fmu/tools/ensembles/ensemble_well_props.py

```diff
@@ -0,0 +1,616 @@
+
+        if "range" in self.config["ensemble"]["realizations"]:
+            start, stop = self.config["ensemble"]["realizations"]["range"].split("-")
+            self.reals = list(range(int(start), int(stop) + 1))
+        elif "entries" in self.config["ensemble"]["realizations"]:
+            self.reals = self.config["ensemble"]["realizations"]["entries"]
+        else:
+            raise ValueError("For 'realizations': 'range' or 'entries' is missing")
+
+        self.itera = self.config["ensemble"]["iteration"]
+
+        self.wellfile = self.config["well"]["file"]
+        self.lognames = self.config["well"].get("lognames", "all")
+        self.mdlog = self.config["well"].get("mdlog", "MDepth")
+        self.mdranges = self.config["well"]["mdranges"]
+        self.welldelta = self.config["well"].get("delta", 2)
+
+        cfgprop = self.config["gridproperties"]
+        self.gridfilestub = Path(cfgprop["grid"]["filestub"])
+        self.gridreuse = cfgprop["grid"].get("reuse_grid", False)
+
+        self.proplist = []
+        props = cfgprop["properties"]
+        for prop in props:
+            name = prop["name"]
+            filestub = prop["filestub"]
+            discrete = prop.get("discrete", False)
+
+            propcase = PropsData(name, filestub, discrete)
+            self.proplist.append(propcase)
+
+        rpt = self.config["report"]
+        if "average_logs" in rpt:
+            self.report_avg_file = rpt["average_logs"].get("fileroot", None)
+            if self.report_avg_file:
+                self.report_avg_file += ".rmswell"
+
+        if "cumulative_lengths" in rpt:
+            self.report_cum_file = rpt["cumulative_lengths"].get("fileroot", None)
+            crit = rpt["cumulative_lengths"].get("criteria", None)
+            if crit is None:
+                raise ValueError("Criteria is missing for 'cumulative_length' option")
+
+            for propcase in self.proplist:
+                if crit and propcase.name in crit:
+                    codes = crit[propcase.name].get("codes", None)
+                    interval = crit[propcase.name].get("interval", None)
+                    if not codes and propcase.discrete:
+                        raise ValueError(
+                            f"Discrete property {propcase.name} must have "
+                            "'codes' as criteria"
+                        )
+                    if not interval and not propcase.discrete:
+                        raise ValueError(
+                            f"Continuous property {propcase.name} must have "
+                            "'interval' as criteria"
+                        )
+
+                    propcase.codes = self._validate_codes(codes)
+                    propcase.interval = self._validate_interval(interval)
+
+        self.report_keep_intermediate = rpt.get("keep_intermediate_logs", False)
+        self.report_show_in_terminal = rpt.get("show_in_terminal", True)
+
+        self.config = None
```

Comment:
Is this line really needed? It creates some mypy complaints since the config is only type hinted as a dict.

Reply:
Don't think so, so I will remove